src/subscription_manager/cli_command/register.py

```diff
@@ -209,6 +212,43 @@ def _do_auto_attach(self, consumer):
             log.exception("Auto-attach failed")
             raise
 
+    def _upload_profile_blocking(self, consumer: dict) -> None:
+        """
+        Try to upload DNF profile to server
+        """
+        with LiveStatusMessage(_("Uploading DNF profile")):
+            try:
+                profile_mgr = inj.require(inj.PROFILE_MANAGER)
+                # 767265: always force an upload of the packages when registering
+                profile_mgr.update_check(self.cp, consumer["uuid"], True)
+            except RemoteServerException as err:
+                # When it is not possible to upload profile ATM, then print only error about this
+                # to rhsm.log. The rhsmcertd will try to upload it next time.
+                log.error("Unable to upload profile: {err}".format(err=str(err)))
+
+    def _upload_profile(self, consumer: dict) -> None:
+        """
+        Try to upload DNF profile to server, when it is supported by server. This method
+        tries to "outsource" this activity to rhsmcertd first. When it is not possible due to
+        various reasons, then we try to do it ourselves in blocking way.
+        """
+        # First try to get PID of rhsmcertd from lock file
+        try:
+            with open("/var/lock/subsys/rhsmcertd", "r") as lock_file:
+                rhsmcertd_pid = int(lock_file.readline())
+        except Exception as err:
```

Comment:
hmm that is quite broad; would it be possible to catch more specific exceptions instead?